components/builder-core/src/api_client.rs

```diff
@@ -189,7 +187,7 @@ impl ApiClient {
 
         debug!("Writing to {}", &tmp_file_path.display());
         let mut f = File::create(&tmp_file_path).map_err(Error::IO)?;
-        io::copy(&mut resp, &mut f).map_err(Error::IO)?;
+        f.write_all(&resp.bytes().await?).map_err(Error::IO)?;
```

Comment:
Just a heads up this is potentially a blocking call. It is probably fine. I missed stuff like this all over the place in the supervisor :sweat-smile:.

Reply:
Addressed in a33367f0febdd930dfb5cba3892af203fc5b3a51